core/src/main/java/com/novoda/espresso/ViewTestRule.java

```diff
@@ -1,50 +1,65 @@
 package com.novoda.espresso;
 
 import android.app.Instrumentation;
-import android.content.Intent;
 import android.support.annotation.LayoutRes;
 import android.support.test.InstrumentationRegistry;
 import android.support.test.rule.ActivityTestRule;
 import android.view.View;
+import android.view.ViewGroup;
 
-public class ViewTestRule<T extends View> extends ActivityTestRule<ViewActivity> {
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+
+public class ViewTestRule<T extends View> extends ActivityTestRule<EmptyActivity> {
 
     private final Instrumentation instrumentation;
+    private final ViewCreator<T> viewCreator;
+
+    private T view;
+
+    public ViewTestRule(@LayoutRes int layoutId) {
+        this(new InflateFromXmlViewCreator<T>(layoutId));
+    }
 
-    @LayoutRes
-    private final int id;
+    public ViewTestRule(ViewCreator<T> viewCreator) {
+        this(InstrumentationRegistry.getInstrumentation(), viewCreator);
+    }
 
-    public ViewTestRule(@LayoutRes int id) {
-        super(ViewActivity.class);
-        this.id = id;
-        instrumentation = InstrumentationRegistry.getInstrumentation();
+    public ViewTestRule(Instrumentation instrumentation, ViewCreator<T> viewCreator) {
+        super(EmptyActivity.class);
+        this.instrumentation = instrumentation;
+        this.viewCreator = viewCreator;
     }
 
     @Override
-    protected Intent getActivityIntent() {
-        Intent intent = new Intent(Intent.ACTION_MAIN);
-        intent.putExtra(ViewActivity.EXTRA_LAYOUT_ID, id);
-        return intent;
+    protected void afterActivityLaunched() {
+        super.afterActivityLaunched();
+        final EmptyActivity activity = getActivity();
+        view = viewCreator.createView(activity, (ViewGroup) activity.findViewById(android.R.id.content));
+
+        runOnMainSynchronously(new Runner<T>() {
+            @Override
+            public void run(T view) {
+                ViewGroup.LayoutParams matchParent = new ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT);
+                activity.setContentView(view, matchParent);
```

Comment:
I'm probably missing something here. But why force `MATCH_PARENT`? In the `ViewCreator` you have the container view, so you can select the desired `LayoutParams` there and set them to the view.

Reply:
:+1: will test with this removed

Reply:
commit 36eeda4de2aabd0e9d3361f826bfcc5ec4791cb0
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Nov 7 17:27:03 2017 +0000

    Avoid forcing layout params


commit 2c268408d0b72a78f8887665e780b3b38208cca1
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Nov 7 17:33:02 2017 +0000

    Add layout params explicitly when setting content view
    
    - otherwise looks like it uses match parent regardless